src/Utility/Point.h

```diff
@@ -0,0 +1,101 @@
+// $Id$
+//==============================================================================
+//!
+//! \file Point.h
+//!
+//! \date Sep 19 2019
+//!
+//! \author Knut Morten Okstad / SINTEF
+//!
+//! \brief Representation of points in 3D space with coordinates and parameters.
+//!
+//==============================================================================
+
+#ifndef UTL_POINT_H
+#define UTL_POINT_H
+
+#include "Vec3.h"
+
+
+namespace utl
+{
+  /*!
+    \brief Class for representing points in 3D space.
+    \details The only difference between this class and the parent class Vec4
+    is that it keeps the spline parameters in an internal array.
+  */
+
+  class Point : public Vec4
+  {
+    Real par[3]; //!< Internal storage of spline parameters
```

Comment:
if you use std::array you don't have to write any copy constructor or assignment operators, it will generate just fine then.

Reply:
Probably, but then there is the `u` pointer in the parent class Vec4. It should not just be copied. In case of this sub-class I want it to point to `par` instead.

Reply:
Was referring to the point(const point&)

Reply:
I know, but in my understanding the generated copy constructor will do `this->u = X.u` while what I do want is `this->u = this->par`. Therefore the explicit implementation.